test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs

```diff
@@ -1677,5 +1677,140 @@ public void UpdateCommand_Self_FailsWithMoreThanOneSource()
+                        "proj2",
+                        packages));
+
+                Util.CreateFile(
+                    projectDirectory1,
+                    "packages.config",
+                    Util.GetNFPackageConfig(packages));
+
+                List<string> projectList = new string[] { "proj1", "proj2" }.ToList();
+
+                // create solution file
+                Util.CreateFile(solutionDirectory, "a.sln",
+                    Util.CreateNFSolutionFileContent(projectList));
+
+                // get paths for projects and solutions
+                string projectFile1 = Path.Combine(projectDirectory1, "proj1.nfproj");
+                string projectFile2 = Path.Combine(projectDirectory2, "proj2.nfproj");
+                string solutionFile = Path.Combine(solutionDirectory, "a.sln");
+
+                var testNuGetProjectContext = new TestNuGetProjectContext();
+                string msbuildDirectory = MsBuildUtility.GetMsBuildToolset(null, null).Path;
+                var projectSystem1 = new MSBuildProjectSystem(msbuildDirectory, projectFile1, testNuGetProjectContext);
+                var projectSystem2 = new MSBuildProjectSystem(msbuildDirectory, projectFile2, testNuGetProjectContext);
+                var msBuildProject1 = new MSBuildNuGetProject(projectSystem1, packagesDirectory, projectDirectory1);
+                var msBuildProject2 = new MSBuildNuGetProject(projectSystem2, packagesDirectory, projectDirectory2);
+
+                using (FileStream stream = File.OpenRead(a1Package))
+                {
+                    var downloadResult = new DownloadResourceResult(stream, packagesSourceDirectory);
+                    await msBuildProject1.InstallPackageAsync(
+                        a1,
+                        downloadResult,
+                        testNuGetProjectContext,
+                        CancellationToken.None);
+                }
+
+                using (FileStream stream = File.OpenRead(b1Package))
+                {
+                    var downloadResult = new DownloadResourceResult(stream, packagesSourceDirectory);
+                    await msBuildProject2.InstallPackageAsync(
+                        b1,
+                        downloadResult,
+                        testNuGetProjectContext,
+                        CancellationToken.None);
+                }
+
+                projectSystem1.Save();
+                projectSystem2.Save();
+
+                var args = new[]
+                {
+                    "update",
+                    solutionFile,
+                    "-Source",
+                    packagesSourceDirectory
+                };
+
+                CommandRunnerResult r = CommandRunner.Run(
+                    Util.GetNuGetExePath(),
+                    workingPath,
+                    string.Join(" ", args),
+                    waitForExit: true);
+
+                Assert.True(r.Item1 == 0, "Output is " + r.Item2 + ". Error is " + r.Item3);
```

Comment:
Sorry I missed this in the first pass, but rather than `Item1`, `Item2`, etc, there are counterparts of `ExitCode`, `Success`, `AllOutput`, etc. I don't know how the team feels, but I prefer the named variables as something like "Item1" isn't obvious to me.
I've been mentioning this in PR reviews when I see them, but it's not documented anywhere. 😕

Eventually, in another PR, maybe we'll finally remove the numbered ones...

Reply:
Fiy: Please use [valueTuple](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples) instead of numbered tuple for code readability.

Reply:
> Fiy: Please use [valueTuple](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples) instead of numbered tuple for code readability.

That's already fixed...